discojs/discojs-core/src/dataset/data/preprocessing/tabular_preprocessing.ts

```diff
@@ -18,10 +18,16 @@ interface TabularEntry extends tf.TensorContainerObject {
 const sanitize: PreprocessingFunction = {
   type: TabularPreprocessing.Sanitize,
   apply: (entry: tf.TensorContainer, task: Task): tf.TensorContainer => {
-    const { xs, ys } = entry as TabularEntry
-    return {
-      xs: xs.map(i => i === undefined ? 0 : i),
-      ys: ys
+    // if preprocessing a dataset without labels, then the entry is an array of numbers
+    if (entry instanceof Array) {
```

Comment:
in case of other `Array` implementation that might happen cross-platform, `instanceof` will fail. is it [recommended to use `Array.isArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs._array.isarray)

```suggestion
    if (Array.isArray(entry)) {
```